### PR TITLE
Replacing Return with GoTo while inlining function calls

### PR DESCRIPTION
**Pull request by czgdp1807**
Closes https://github.com/lcompilers/lpython/issues/519

### CODE REVIEW

**Comment by czgdp1807** [comment]
Will fix the git history later.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -2841,6 +2841,13 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_BlockCall(const ASR::BlockCall_t& x) {
+        if( x.m_label != -1 ) {
+            if( llvm_goto_targets.find(x.m_label) != llvm_goto_targets.end() ) {
```
Comment (czgdp1807) [inline src/libasr/codegen/asr_to_llvm.cpp:2845]: Fixed. This was the mistake, `!=` replaced with `==`.

**Comment by czgdp1807** [comment]
This is ready for review. If you approve this then feel free to do so but do not merge. I will apply the same changes to LFortran and make sure that they work with function inlining for Fortran as well. Once both of the MRs pass all the tests then we will merge them simultaneously.

**Comment by czgdp1807** [comment]
@certik Please feel free to review. Nothing else to be done here now.

**Comment by czgdp1807** [comment]
@certik Can we merge this too? The same changes have been merged in LFortran.

**Review by certik — APPROVED** [review]
Yes, go ahead.

**Pull request merged**